qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1520A.py

```diff
@@ -161,6 +608,216 @@ def abort(self):
+
+            volt_monitor: Boolean
+
+        """
+
+        self.adc_mode(adc_mode)
+        self.adc_coef(adc_coef)
+        self.frequency(freq)
+        self.voltage_ac(ac_rms)
+        self.cv_sweep.sweep_auto_abort(abort_enabled)
+        self.cv_sweep.post_sweep_voltage_cond(post_sweep_voltage_cond)
+        self.cv_sweep.hold(hold_delay)
+        self.cv_sweep.delay(delay)
+        self.cv_sweep.step_delay(step_delay)
+        self.cv_sweep.trigger_delay(trigger_delay)
+        self.cv_sweep.measure_delay(measure_delay)
+        self.sweep_mode(sweep_mode)
+        self.sweep_start(v_start)
+        self.sweep_end(v_end)
+        self.sweep_steps(n_steps)
+        self.measurement_mode(constants.MM.Mode.CV_DC_SWEEP)
+        self.impedance_model(imp_model)
+        self.ac_dc_volt_monitor(volt_monitor)
+        self.ranging_mode(ranging_mode)
+        self.measurement_range_for_non_auto(fixed_range_val)
+
+        err = self.root_instrument.error_message()
+        if err == '+0,"No Error."':
+            self.setup_fnc_already_run = True
+        self.self.root_instrument.clear_buffer_of_error_message()
+        return err
+
+    @staticmethod
+    def parse_sweep_data(raw_data: str) -> tuple:
+        no_commas = raw_data.split(',')
+        no_str = [float(val[3:]) for val in no_commas]
+        param1 = []
+        param2 = []
+        for i, val in enumerate(no_str):
+            if i % 2:
+                param2.append(val)
+            else:
+                param1.append(val)
+
+        param1 = list(np.array(param1))
+        param2 = list(np.array(param2))
+        return param1, param2
+
+
+class CVSweepMeasurement(MultiParameter):
+    """
+    CV sweep measurement outputs a list of primary (capacitance) and secondary
+    parameter (disipation).
+
+    Args:
+        name: Name of the Parameter.
+
+        instrument: Instrument to which this parameter communicates to.
+    """
+
+    def __init__(self, name, instrument, **kwargs):
+        super().__init__(
+            name,
+            names=tuple(['Capacitance', 'Dissipation']),
+            units=tuple(['F', 'unit']),
+            labels=tuple(['Parallel Capacitance', 'Dissipation factor']),
+            shapes=((1,),) * 2,
+            setpoint_names=(('Voltage',),) * 2,
+            setpoint_labels=(('Voltage',),) * 2,
+            setpoint_units=(('V',),) * 2,
+            **kwargs)
+        self._instrument = instrument
+
+    def get_raw(self):
+        if not self._instrument.setup_fnc_already_run:
+            raise Warning('Sweep setup has not yet been run successfully')
+        num_steps = self._instrument.sweep_steps()
+        delay_time = self._instrument.cvsweep.step_delay()
+
+        self.shapes = ((num_steps,),) * 2
+        self.setpoints = ((self._instrument.cv_sweep_voltages(),),) * 2
+
+        fudge = 6  # fudge factor for setting timeout
+        estimated_measurement_time = delay_time * num_steps  # the calculation can be improved
```

Comment:
For PLC ADC mode, it seems that the total time the xe() command is blocking is 2 * NPLC * Nsteps + 25ms if 2*NPLC + 25ms is larger than step_delay.  If step_delay is set longer, then step_delay is the limiting time.  I think something from my original measurement time calculation got lost in the transfer from my fork to this PR.

Reply:
I have some plots of the measurement time I can share offline.

Reply:
Thanks for looking out this John. I will revert it back to the calculation you wrote.

Reply:
Fixed this as per you suggestion